state-chain/runtime/src/migrations.rs

```diff
@@ -0,0 +1,50 @@
+//! Chainflip runtime storage migrations.
+use crate::System;
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+use sp_std::{marker::PhantomData, vec::Vec};
+
+pub mod threshold_signature_callbacks;
+
+#[cfg(feature = "try-runtime")]
+use sp_runtime::DispatchError;
+
+//const SPEC_VERSION: u32 = 101;
+
+/// A runtime storage migration that will only be applied if the `SPEC_VERSION` matches the
+/// post-upgrade runtime's spec version.
+pub struct VersionedMigration<U, const SPEC_VERSION: u32>(PhantomData<U>);
+
+impl<U, const SPEC_VERSION: u32> OnRuntimeUpgrade for VersionedMigration<U, SPEC_VERSION>
+where
+	U: OnRuntimeUpgrade,
+{
+	fn on_runtime_upgrade() -> Weight {
+		if System::runtime_version().spec_version == SPEC_VERSION {
+			U::on_runtime_upgrade()
+		} else {
+			log::info!(
+				"Skipping storage migration for version {:?} - consider removing this from the runtime.",
+				SPEC_VERSION
+			);
+			Weight::zero()
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+		if System::runtime_version().spec_version == SPEC_VERSION {
+			U::pre_upgrade()
+		} else {
+			Ok(vec![])
+		}
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(state: Vec<u8>) -> Result<(), DispatchError> {
+		if System::runtime_version().spec_version == SPEC_VERSION {
```

Comment:
Is this correct (`System::runtime_version().spec_version == SPEC_VERSION` in each of these methods)? Shouldn't some of them be `<` or `>` ?

Reply:
should be correct. This migration will only run if the post runtime upgrade spec version matches exactly to SPEC_VERSION for which this migration is created for as says in the comment. we then initialise this versioned migration with the spec version we want it to apply to (one plus the current) in the runtime lib file.